src/jinja2/compiler.py

```diff
@@ -623,6 +627,14 @@ def write_commons(self):
         """
         self.writeline("resolve = context.resolve_or_missing")
         self.writeline("undefined = environment.undefined")
+        # make sure in the root render function the correct escape is used
+        self.writeline("escape = context.eval_ctx.get_escape_function()")
+        # make sure in the root render function the coorect Markup
+        # class ise used
```

Comment:
```suggestion
        # class is used
```

but maybe these comments here are a bit too verbose?